apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.tsx

```diff
@@ -45,20 +44,33 @@ export interface CheckboxDropdownProps
   checkedOptions: string[];
   /** CheckboxDropdown onChange handler */
   onChange: (args: React.ChangeEvent<HTMLInputElement>) => void;
-  /** CheckboxDropdown onSelectAll handler */
+}
+
+interface CheckboxDropdownWithoutControlProps
+  extends BaseCheckboxDropdownProps {
+  showControls: false;
+}
+
+interface CheckboxDropdownWithControlsProps extends BaseCheckboxDropdownProps {
+  showControls: true;
```

Comment:
Can we replace this with hideControls instead?
We need to ensure that default behaviour remains the same and controls are shown by default.
Using `showControls` instead of `hideControls` will make it much easier to unintentionally forget to show controls

Reply:
As I understand it, there's basically no default behavior:
- if you don't include a showControls prop, you get TS errors
- if you include `showControls=true`, you'll get TS errors if you don't include the select/clear all click handlers and button text
- if you include showControls=false, you don't need to (/will get an error if you do) include the click handlers or button text props

All of that to say, I don't think there's a default that will allow you to accidentally forget to include the controls.

I used `showControls` rather than `hideControls` to avoid the double negative in code (ie, `showControls &&` rather than `!hideControls &&` is easier for my brain)

Reply:
That makes sense to me, updated!

Interestingly, learned that `hideControls?: false` is different than `hideControls: false | undefined`

Reply:
> Interestingly, learned that hideControls?: false is different than hideControls: false | undefined

Oh yeah, thanks for bringing this up! Just googled it and yeah, pretty smart from TS